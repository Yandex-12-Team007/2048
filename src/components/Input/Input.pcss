.input-container {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  &__label {
    font-size: 10px;
    color: var(--gray);
  }
}

.input {
  border: none;
  border-bottom: 1px solid #cdc1b3;
  padding: 7px 0;
  font-size: 14px;
  background-color: transparent;
  color: var(--main-text-color);

  &:focus-visible {
    outline: none;
  }

  &__error-message {
    position: absolute;
    right: 0;
    bottom: 0;
    font-size: 10px;
    color: var(--error-text-color);
    z-index: 2;
  }
}
