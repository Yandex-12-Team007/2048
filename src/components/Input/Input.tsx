import classNames from 'classnames';
import React, {forwardRef, InputHTMLAttributes, RefObject} from 'react';
import './Input.pcss';

interface IInputProps extends InputHTMLAttributes<HTMLInputElement> {
  id: string;
  label?: string;
  type?: string;
  className?: string;
  errorMessage?: string;
  errorClassName?: string;
}

const Input: React.FunctionComponent<IInputProps> = forwardRef((props, ref) => {
  const {
    id,
    className,
    label,
    type,
    errorMessage,
    errorClassName,
    ...restProps
  } = props;

  return (
    <div className='input-container'>
      {
        label && (
          <label
            className='input-container__label'
            htmlFor={id}
          >
            {label}
          </label>
        )
      }
      <input
        ref={ref as RefObject<HTMLInputElement>}
        id={id}
        className={classNames('input', className)}
        type={type || 'text'}
        {...restProps}
      />
      <span className={classNames('input__error-message', errorClassName)}>
        {errorMessage}
      </span>
    </div>
  )
});

export default Input;
