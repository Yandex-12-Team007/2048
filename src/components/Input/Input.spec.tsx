import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

const TEST_ID = 'name';
const TEST_LABEL = 'label name';
const TEST_CLASS = 'test_class';
const ERROR_MESSAGE = 'error message';
const ERROR_CLASS = 'test_error_class';
const TYPE = 'text';

describe('Test Input', () => {
  it('check props', () => {
    const render = renderer.create(
        <Input
          id={TEST_ID}
          label={TEST_LABEL}
          type={TYPE}
          className={TEST_CLASS}
          errorClassName={ERROR_CLASS}
          errorMessage={ERROR_MESSAGE}
        />
    ).toJSON();

    expect(render).toMatchSnapshot();
  });
})
