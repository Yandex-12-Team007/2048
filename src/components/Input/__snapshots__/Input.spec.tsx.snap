// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="input-container"
>
  <label
    className="input-container__label"
    htmlFor="name"
  >
    label name
  </label>
  <input
    className="input test_class"
    id="name"
    type="text"
  />
  <span
    className="input__error-message test_error_class"
  >
    error message
  </span>
</div>
`;

exports[`Test Input check props 1`] = `
<div
  className="input-container"
>
  <label
    className="input-container__label"
    htmlFor="name"
  >
    label name
  </label>
  <input
    className="input test_class"
    id="name"
    type="text"
  />
  <span
    className="input__error-message test_error_class"
  >
    error message
  </span>
</div>
`;
